crates/router/src/core/payments/helpers.rs

```diff
@@ -5669,6 +5666,37 @@ where
     Ok(())
 }
 
+#[cfg(feature = "v2")]
+pub async fn override_setup_future_usage_to_on_session<F, D>(
+    db: &dyn StorageInterface,
+    payment_data: &mut D,
+) -> CustomResult<(), errors::ApiErrorResponse>
+where
+    F: Clone,
+    D: payments::OperationSessionGetters<F> + payments::OperationSessionSetters<F> + Send,
+{
+    if payment_data.get_payment_intent().setup_future_usage == enums::FutureUsage::OffSession {
```

Comment:
Variant of FutureUsage are changed to `CIT` `MIT` `OneTime`

Reply:
will take this up in a separate PR.